src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/service/PlanBulkAlert.kt

```diff
@@ -1,21 +1,77 @@
 package uk.gov.justice.digital.hmpps.hmppsalertsapi.service
 
 import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.client.prisonersearch.PrisonerSearchClient
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.config.AlertRequestContext.Companion.PRISON_NUMBER_REGEX
 import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.BulkPlanRepository
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.PersonSummary
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.PersonSummaryRepository
 import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.getPlan
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.toPersonSummary
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.exceptions.InvalidRowException
 import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.BulkPlan
-import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.BulkRequest
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.AddPrisonNumbers
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.BulkAction
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.SetAlertCode
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.SetDescription
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.repository.AlertCodeRepository
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.repository.getByCode
 import java.util.UUID
 import uk.gov.justice.digital.hmpps.hmppsalertsapi.entity.BulkPlan as Plan
 
+@Transactional
 @Service
-class PlanBulkAlert(private val planRepository: BulkPlanRepository) {
+class PlanBulkAlert(
+  private val planRepository: BulkPlanRepository,
+  private val alertCodeRepository: AlertCodeRepository,
+  private val personSummaryRepository: PersonSummaryRepository,
+  private val prisonerSearch: PrisonerSearchClient,
+) {
   fun createNew(): BulkPlan = planRepository.save(Plan()).toModel()
-  fun update(id: UUID, request: BulkRequest): BulkPlan {
+  fun update(id: UUID, action: BulkAction): BulkPlan {
     val plan = planRepository.getPlan(id)
+    when (action) {
+      is SetAlertCode -> plan.setAlertCode(action)
+      is SetDescription -> plan.setDescription(action)
+      is AddPrisonNumbers -> plan.addPrisonNumbers(action)
+    }
 
     return plan.toModel()
   }
+
+  private fun Plan.setAlertCode(action: SetAlertCode) {
+    alertCode = alertCodeRepository.getByCode(action.alertCode)
+  }
+
+  private fun Plan.setDescription(action: SetDescription) {
+    description = action.description
+  }
```

Comment:
UI might need to set description to null.

eg, user starts with CSIP alertCode and provides a description text.
these are submitted to backend to get estimated number of alerts to create/update.

then user might go back to change alertCode to DOCGM, and unset description text.